### ISSUE TITLE
Cursor plugin causes word breaking

### ISSUE THREAD

**Issue by bkempe** [yjs/y-prosemirror#97]
**Checklist**

* [x] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [x] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
The cursor plugin breaks words which can be observed especially when the cursor is placed in long words and using a smaller width of the text space.

**To Reproduce**
1. Open two browser windows with width 800px
2. In both windows, go to https://demos.yjs.dev/prosemirror/prosemirror.html
3. Paste "Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Donaudampfschiffahrtsgesellschaftskapitän" in one window
4. Place the cursor after Donaudampf in one window
5. See that the word gets broken in the _other_ window.

The DOM looks like this:
```html
<p>Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Test test 123 Donaudampf<span 
class="ProseMirror-yjs-cursor ProseMirror-widget" style="border-color: #ffa500" contenteditable="false"><div 
style="background-color: #ffa500">User: 255092616</div></span>schiffahrtsgesellschaftskapitän</p>
```

**Expected behavior**
Words should not get broken. Example: https://slate-yjs.dev/remote-cursors-overlay

**Screenshots**
<img width="821" alt="Screen Shot 2022-03-21 at 20 27 06" src="https://user-images.githubusercontent.com/568972/159349147-971d45fc-3d34-447f-9fc3-83baa2ca7f09.png">

**Additional context**

**[We are a Sponsor :)](https://github.com/vowel-com)**

**Comment by bkempe** [comment]
@dmonad Thanks for looking into this! One compromise solution may be to only wrap URLs(which are the most likely items to be long in a typical notepad) in a block.

**Comment by dmonad** [comment]
That makes sense. You can basically set them in a `<div style="white-space: nowrap;">` element.

**Comment by dmonad** [comment]
There's now a solution to this problem. See https://github.com/yjs/y-codemirror.next/pull/12

I'll be porting the solution to y-prosemirror (probably in the next ~two weeks). But if somebody wants to give this a try, that would be great!

**Comment by bkempe** [comment]
@dmonad BTW, this is how Atlassian Atlas, which also uses Prosemirror, does it: https://bitbucket.org/atlassian/atlassian-frontend-mirror/src/f388d30ecaa34cf83e8d8a55167b800c9f027db9/editor/editor-core/src/plugins/collab-edit/utils.ts?at=master#lines-105

**Comment by dmonad** [comment]
This issue is finally fixed in `y-prosemirror@v1.0.18` :tada:
